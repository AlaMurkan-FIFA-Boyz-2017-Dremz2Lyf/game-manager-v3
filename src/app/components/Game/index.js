import React from 'react';
import PropTypes from 'prop-types';
import { Segment, Header, Container, Grid, Label, Image } from 'semantic-ui-react';
import GameForm from '../GameForm';

const Game = ({ status }) => (
  status === 'started' ? (
    <Segment>
      <Container>
        <GameForm status={status} />
        <Segment.Group size="small" horizontal>
          <Segment textAlign="left">
            <Grid>
              <Grid.Row>
                <Grid.Column mobile={16} computer={10}>
                  <Header>
                  Scott
                  </Header>
                Tottenham Hotspur
                  <Label>
                  4-3-2
                  </Label>
                </Grid.Column>
                <Grid.Column only="computer tablet" width={6}>
                  <Image
                    floated="right"
                    size="mini"
                    src="http://upload.wikimedia.org/wikipedia/de/b/b4/Tottenham_Hotspur.svg"
                  />
                </Grid.Column>
              </Grid.Row>
            </Grid>
          </Segment>
          <Segment textAlign="right">
            <Grid>
              <Grid.Row>
                <Grid.Column only="computer tablet" width={6}>
                  <Image
                    size="mini"
                    src="http://upload.wikimedia.org/wikipedia/de/d/da/Manchester_United_FC.svg"
                  />
                </Grid.Column>
                <Grid.Column mobile={16} computer={10}>
                  <Header>
                  Agustin
                  </Header>
                  Manchester United FC
                  <Label>
                  0-3-5
                  </Label>
                </Grid.Column>
              </Grid.Row>
            </Grid>
          </Segment>
        </Segment.Group>
      </Container>
    </Segment>
  ) : (
    <Segment>
      <Container>
        <GameForm status={status} />
        <Segment.Group size="small" horizontal>
          <Segment compact textAlign="left">
            <Grid>
              <Grid.Row>
                <Grid.Column mobile={16} computer={10}>
                  <Header>
                  Scott
                  </Header>
                  <Label>
                  4-3-2
                  </Label>
                </Grid.Column>
                <Grid.Column only="computer tablet" width={6}>
                  <Image
                    floated="right"
                    size="mini"
                  />
                </Grid.Column>
              </Grid.Row>
            </Grid>
          </Segment>
          <Segment compact textAlign="right">
            <Grid>
              <Grid.Row>
                <Grid.Column only="computer tablet" width={6}>
                  <Image
                    size="mini"
                  />
                </Grid.Column>
                <Grid.Column mobile={16} computer={10}>
                  <Header>
                  Agustin
                  </Header>
                  <Label>
                  0-3-5
                  </Label>
                </Grid.Column>
              </Grid.Row>
            </Grid>
          </Segment>
        </Segment.Group>
      </Container>
    </Segment>
  )
);

Game.propTypes = {
  status: PropTypes.string,
};

export default Game;
