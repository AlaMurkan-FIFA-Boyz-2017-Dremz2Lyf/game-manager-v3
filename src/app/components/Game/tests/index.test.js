import { shallow } from 'enzyme';
import React from 'react';

import Game from '../';
import { created, started } from './fixtures';

describe('<Game />', () => {
  it('should match the snapshot for created games', () => {
    const { home, away } = created;
    const wrapper = shallow(<Game home={home} away={away} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot for started games', () => {
    const { home, away } = started;
    const wrapper = shallow(<Game home={home} away={away} />);
    expect(wrapper).toMatchSnapshot();
  });
});
