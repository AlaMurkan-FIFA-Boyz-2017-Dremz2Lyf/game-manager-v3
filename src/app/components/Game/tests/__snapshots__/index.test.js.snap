// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Game /> should match the snapshot for created games 1`] = `
<Segment>
  <GameForm />
  <ItemGroup>
    <Item
      className="styles__player-38Wf0"
    >
      <ItemContent>
        <ItemHeader>
          Agustin
        </ItemHeader>
      </ItemContent>
    </Item>
    <Item
      className="styles__player-38Wf0"
    >
      <ItemContent>
        <ItemHeader>
          Scott
        </ItemHeader>
      </ItemContent>
    </Item>
  </ItemGroup>
</Segment>
`;

exports[`<Game /> should match the snapshot for started games 1`] = `
<Segment>
  <GameForm />
  <ItemGroup>
    <Item
      className="styles__player-38Wf0"
    >
      <ItemContent>
        <ItemHeader>
          Scott
        </ItemHeader>
        <ItemDescription>
          Tottenham Hotspur
        </ItemDescription>
      </ItemContent>
    </Item>
    <Item
      className="styles__player-38Wf0"
    >
      <ItemContent>
        <ItemHeader>
          Agustin
        </ItemHeader>
        <ItemDescription>
          The Manure
        </ItemDescription>
      </ItemContent>
    </Item>
  </ItemGroup>
</Segment>
`;
